modules/commandModule/commandWorker_flight.py

```diff
@@ -1,18 +1,56 @@
-from modules.commandModule.commandFns import write_pigo
+from modules.commandModule.commandModule import CommandModule
+from modules.commandModule.commandFns import write_pigo, read_pogi
 
-def flight_command_worker(pipelineIn, pipelineOut):
-	# pipelineIn gives [[x,y], [range]]
-	if pipelineIn.empty:
-	   continue
-	
-	data = pipelineIn.get()
-	# Cache data here
-	newPigo = {
-		'gpsCoordinates': [x, y]
-	}
+POGI_DIR = ""
+PIGO_DIR = ""
 
-	write_pigo(newPigo)
+def pogi_subworker(pipelineOut, POGI_DIR):
 
-	#TODO: Write logic to get POGI
+    # get the pogi data
+    changed, pogiData = read_pogi(POGI_DIR)
 
-	
\ No newline at end of file
+    # if pogi has changed, output to the pipeline
+    if changed:
+        pipelineOut.put(pogiData)
+
+
+def flight_command_worker(pipelineIn, pipelineOut, pause, exitRequest):
+    
+    """
+    Worker function for PIGO and POGI data transactions
+
+    Parameters
+    ----------
+    pipelineIn : multiprocessing.Queue
+        input pipeline for pigoData
+    pipelineOut : multiprocessing.Queue
+        output pipeline for pogiData
+    pause : multiprocessing.Lock
+        a lock shared between worker processes
+    exitRequest : multiprocessing.Queue
+        a queue that determines when the worker process stops
+    """
+
+    while True:
+
+        pause.acquire()
+        pause.release()
+        
+        # PIGO Logic
+        # pipelineIn gives [[x,y], [range]]
+        if pipelineIn.empty:
```

Comment:
This means if the in pipeline is empty we will have to wait to get in PIGO data. The POGI logic should be moved above the PIGO logic, so that we make sure we're not delaying the flow of telemetry data in.